### ISSUE TITLE
Disable transitions API to support unit testing

### ISSUE THREAD

**Issue by rbu** [d3/d3#1789]
Transitions with their unpredictable timing make it hard for us to test changes in an SVG with JS unit tests.
We've tried mocking the wall clock and calling d3.timer.flush(), but for some reason not even that helps (unless you flush the timer a few ten thousand times).
Looking at https://stackoverflow.com/questions/20068497/d3-transition-in-unit-testing and https://stackoverflow.com/questions/14443724/disabling-all-d3-animations-for-testing I suppose I'm not the only one who has this problem, so I propose:

There should be a way to globally disable all transitions and animations and move to the final state immediately, similar to http://api.jquery.com/jQuery.fx.off/

**Comment by skhilko** [comment]
@mbostock, could you suggest how to skip the animation in d3 v4? None of the solutions above seem to work for me.

**Comment by cvkline** [comment]
Same observation…  winding time forward to Infinity no longer works to flush D3 transitions in version 4.  I'm using asynchronous Mocha tests and using a setTimeout to wait for it to complete, but that does have the potential to slow down our testing environment.

**Comment by larsenmtl** [comment]
@skhilko @cvkline Version 4 moved to [performance.now](https://github.com/d3/d3-timer/blob/master/src/timer.js#L10), so you need to use  ` performance.now = function() { return Infinity; }; d3.timerFlush();`

**Comment by classiemilio** [comment]
@larsenmtl @skhilko @cvkline I tried that and was running into problems with `timerFlush` using a cached value for `performance.now()` ([clockNow here](https://github.com/d3/d3-timer/blob/master/src/timer.js#L14)). To get around it, I made a method that I can call to artificially move the clock a certain amount and run the transitions that should be ran. I use promises with Mocha for testing and can chain a few calls to this if I want to test chained transitions.

```js
clockTick(milliseconds) {
  const currentNow = performance.now();

  performance.now = () => currentNow + milliseconds;

  // The new animation frame means d3's timers will check performance.now() again
  return new Promise(resolve => requestAnimationFrame(resolve));
}
```
